File: euler001/euler1.js

function is_multiple(num, divs) {
    'use strict';
    for (var i=0; i<divs.length; i++) {
        if (num % divs[i] === 0) {
            return true;
        }
    }
    return false;
}

function euler1(nmax, divs) {
    'use strict';
    var total = 0;

    for (var i=1; i<nmax; i++) {
        if (is_multiple(i, divs)) {
            total = total + i;
        }
    }

    return total;
}

console.log(euler1(1000, [3, 5]));

File: euler001/euler1.py
#!/usr/bin/python

def is_multiple(num, divs):
    for div in divs:
        if num % div == 0:
            return True
    return False

def multiples(max_number, divs):
    return [i for i in range(max_number) if is_multiple(i, divs)]

def sum_multiples(max_number, divs):
    nums = multiples(max_number, divs)
    total = 0
    for i in nums:
        total = total + i
    return total 

if __name__ == '__main__':
    max_number = 1000
    divs = [3, 5]
    print(sum_multiples(max_number, divs))
